apollo-router/src/warp_http_server_factory.rs

```diff
@@ -676,4 +692,36 @@ mod tests {
             .iter()
             .for_each(|accepts| assert!(!prefers_html(accepts.to_string())));
     }
+
+    #[test(tokio::test)]
+    async fn it_provides_health_status_body() -> Result<(), FederatedServerError> {
+        let filter = get_health_request();
+
+        let res = warp::test::request()
+            .path("/.well-known/apollo/server-health")
+            .reply(&filter)
+            .await;
+
+        assert_eq!(res.status(), 200);
+        assert_json_snapshot!(String::from_utf8_lossy(res.body()));
+
+        Ok(())
+    }
+
+    #[test(tokio::test)]
+    async fn it_provides_health_status_header() -> Result<(), FederatedServerError> {
```

Comment:
[nit] Never return errors in test! That's actually a common mistake.

Test in rust are processes that either succeed with an exit status 0 or fails with an exit status non-zero. Therefore all the standard assertion macros are based on panicking: `assert_eq!()` panics when the test fails.

Pancking in test == test failing, this is fine and it is even better to panic because you can get a stacktrace of where it 
went wrong.

So my rule of thumb is:
- use unwrap() a lot in test
- never use `?` and return a result (otherwise you lose the stacktrace)
- use expect() if it the test doesnt give a useful information to the developer (in custom macros for example)

Reply:
I agree with this. I don't return errors from tests usually. I copied this from the existing tests and assumed it was the preferred approach. Your rules of thumb are the same as mine! :)